trio/_core/_io_windows.py

```diff
@@ -763,6 +770,7 @@ def abort(raise_cancel_):
             return _core.Abort.FAILED
 
         await _core.wait_task_rescheduled(abort)
+        info = self._overlapped_entries.pop(lpOverlapped)
```

Comment:
`wait_task_rescheduled` returns the (unwrapped) thing you pass to `reschedule`, so this would be `info = await _core.wait_task_rescheduled(abort)`